File: c/src/Behavioral/Interpreter/DvdTitleExpression.c

//DvdTitleExpression - Three Of Four Terminal Expressions

#include "DvdExpression.h"

char * DvdTitleExpression_interpret( DvdExpression_t * d, DvdInterpreterContext_t *  ctx) 
{
	return List_csv_str( DvdInterpreterContext_getAllTitles( ctx ) );
}

DvdExpression_t * DvdTitleExpression_new()
{
	return DvdExpression_new( DvdTitleExpression_interpret );
}

File: php/src/Behavioral/Interpreter/DvdTitleExpression.php

<?

# //DvdTitleExpression - One Of the Terminal Expressions

require_once 'DvdAbstractExpression.php';
require_once 'DvdInterpreterContext.php';

class DvdTitleExpression extends DvdAbstractExpression
{
	function interpret (DvdInterpreterContext $ctx)
	{
		return implode(', ',$ctx->getAllTitles());
	}
}


